HackerNews/Stores/StoryImageInfoStore.swift

```diff
@@ -10,69 +10,111 @@ import Foundation
                 DispatchQueue.main.async {
-                    completionHandler(cachedStoryImageInfo.touchIcon)
+                    completionHandler(cachedStoryImageInfo.imageIconURL)
                 }
             } else {
-                self.imageInfo(url: url) { (storyImageinfo) in
-                    self.storyImageCache.addStoryImageInfo(storyHost: host, storyImageInfo: storyImageinfo)
-                    completionHandler(storyImageinfo.touchIcon)
+                self.storyImageInfo(url: url) { (storyImageInfo) in
+                    self.storyImageInfoCache.addStoryImageInfo(storyHost: host, storyImageInfo: storyImageInfo)
+                    DispatchQueue.main.async {
+                        completionHandler(storyImageInfo.imageIconURL)
+                    }
                 }
             }
         }
     }
     
     func ogImageURL(url: String, completionHandler: @escaping (URL?) -> Void) {
         queue.async {
-            self.imageInfo(url: url) { (storyImageInfo) in
-                completionHandler(storyImageInfo.ogImage)
-            }
-        }
-    }
-    
-    private func imageInfo(url: String, completionHandler: @escaping (StoryImageInfo) -> Void) {
-        guard let url = URL(string: url) else { return }
-        var storyImageInfo = StoryImageInfo(url: url)
-        self.html(url: url) { (html) in
-            guard let html = html else {
+            guard let url = URL(string: url) else {
                 DispatchQueue.main.async {
-                    completionHandler(storyImageInfo)
+                    completionHandler(nil)
                 }
                 return
             }
-            let document: Document? = try? SwiftSoup.parse(html, url.absoluteString)
-            var elements: [Element] = []
-            for node in [try? document?.select("head"), try? document?.select("body")].compactMap({ $0 }) {
-                guard let metaTags = try? node.select("meta").array(), let linkTags = try? node.select("link").array() else {
+            self.html(url: url) { (html) in
+                guard let html = html else {
+                    DispatchQueue.main.async {
+                        completionHandler(nil)
+                    }
                     return
                 }
-                elements.append(contentsOf: metaTags)
-                elements.append(contentsOf: linkTags)
-                let images = self.images(elements: elements)
-                for (key, (url, _)) in images {
+                var ogImageURL: URL?
+                for element in self.parse(html, url.absoluteString, forTag: .meta) {
+                    if ogImageURL == nil {
+                        ogImageURL = element.ogImageURL
+                    } else {
+                        break
+                    }
+                }
+                DispatchQueue.main.async {
+                    completionHandler(ogImageURL)
+                }
+            }
+        }
+    }
```

Comment:
This change is not to go through link tags rather go through only meta tags where og:image's are declared.